research/context/preselection/preselect_wikipedia_comments.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tqdm import tqdm
+import pandas as pd
+import tensorflow as tf
+
+# Connect to Google's BigQuery
+from google.cloud import bigquery
+client = bigquery.Client(project="wikidetox-viz")
+
+FLAGS = tf.app.flags.FLAGS
+tf.app.flags.DEFINE_integer("zones_num", 10, "Number of zones to be used for sampling.")
+tf.app.flags.DEFINE_string("output_sample_path", None, "Path where the output sample CSV should be written.")
+tf.app.flags.DEFINE_string("output_short_sample_path", None, "Path where the small (100) output sample CSV should be written.")
+
+def query(low_tox=0, high_tox=1, rand=.5, limit=1000, type="ADDITION", max_txt=1000):
+    """
+    Google Big Query sampling data of specific types
+    :param low_tox: comments should be higher than that
+    :param high_tox: comments should be lower than that
+    :param rand: random selection (p=50%)
+    :param limit: fetch some
+    :param type: kind of comments (wikipedia specific)
+    :param max_txt: max length of comment
+    :return: dataframe with results
+    """
+    query = """
+    SELECT * 
+        FROM `wikiconv_v2.en_20180701_external`
+        WHERE length(cleaned_content)>1 AND length(cleaned_content)<{max_txt} AND replyTo_id!="null" AND type="{typ}" AND toxicity>{low} AND toxicity<{high} AND RAND()<{ran}
+        LIMIT {lim}
```

Comment:
Because this code uses the default order, this isn't quite random; it's random 50% of the initial subset. I think we need to order by the random value to get an actual random subset (or use hash, or timestamp tricks as your source of randomness). You way want to create a very non-random table to test this?

Reply:
Tested, you were right. I changed this to an "order+limit" approach.